src/d3a/d3a_core/export.py

```diff
@@ -662,3 +665,74 @@ def _plot_avg_trade_graph(self, stats, area_name, key, label):
+        """
+        new_subdir = os.path.join(subdir, area.slug)
+        self._plot_energy_profile_hr(area, new_subdir)
+        for child in area.children:
+            if child.children:
+                self.plot_energy_trade_profile_hr(child, new_subdir)
+
+    def _plot_energy_profile_hr(self, area: Area, subdir: str):
+        """
+        Plots history of energy trades plotted for each market_slot
+        """
+        barmode = "relative"
+        xtitle = 'Time'
+        ytitle = 'Energy [kWh]'
+        market_name = area.name
+        title = f'HR Energy Trade Profile of {market_name}'
+        plot_dir = os.path.join(self.plot_dir, subdir, "energy_profile_hr")
+        mkdir_from_str(plot_dir)
+        for market_slot_unix, data in area.stats.market_trades.items():
+            market_slot = from_timestamp(market_slot_unix)
+            plot_data = self.add_plotly_graph_dataset(data, market_slot)
+            if len(plot_data) > 0:
+                market_slot_str = market_slot.format(DATE_TIME_FORMAT)
+                output_file = \
+                    os.path.join(plot_dir, f'energy_profile_hr_'
+                                           f'{market_name}_{market_slot_str}.html')
+                time_range = [market_slot - GlobalConfig.tick_length,
+                              market_slot + GlobalConfig.slot_length + GlobalConfig.tick_length]
+                PlotlyGraph.plot_bar_graph(barmode, title, xtitle, ytitle, plot_data, output_file,
+                                           time_range=time_range)
+
+    @staticmethod
+    def add_plotly_graph_dataset(market_trades, market_slot):
+        plotly_dataset_list = []
+        seller_dict = {}
+        buyer_dict = {}
+        # 1. accumulate data by buyer and seller:
+        for timestamp, data in market_trades.items():
+            time = from_timestamp(timestamp, tz=TIME_ZONE)
+            # This zero point is needed to make plotly also plot the first data point:
+            zero_point_dict = {"timestamp": [market_slot - GlobalConfig.tick_length],
+                               "energy": [0.0]}
+            if data["seller"] not in seller_dict:
+                seller_dict[data["seller"]] = deepcopy(zero_point_dict)
+            if data["buyer"] not in buyer_dict:
+                buyer_dict[data["buyer"]] = deepcopy(zero_point_dict)
+
+            seller_dict[data["seller"]]["timestamp"].append(time)
+            seller_dict[data["seller"]]["energy"].append(data["energy"] * ENERGY_SELLER_SIGN_PLOTS)
+            buyer_dict[data["buyer"]]["timestamp"].append(time)
+            buyer_dict[data["buyer"]]["energy"].append(data["energy"] * ENERGY_BUYER_SIGN_PLOTS)
+
+        # 2. Create bar plot objects and collect them in a list
+        for agent, data in seller_dict.items():
+            data_obj = go.Bar(x=data["timestamp"],
+                              y=data["energy"],
+                              width=GlobalConfig.tick_length.seconds * 1000,
```

Comment:
Would be nice to replace the constant factor with any variable

Reply:
Yeah. After long search i found out, that if the xaxis is in datetime format, the widths of the bars can be set by providing a number in milliseconds. Shall I leave a comment about this in the code? I would not want to create a variable in constants.py. Or what are you suggesting?

Reply:
Yes please leave a comment about this in the code.